core/tls-websocket-server/src/ws_server.rs

```diff
@@ -257,18 +289,14 @@ where
 		connection_token: ConnectionToken,
 		message: String,
 	) -> WebSocketResult<()> {
-		let mut connections_lock =
-			self.connections.write().map_err(|_| WebSocketError::LockPoisoning)?;
-		let connection = connections_lock
-			.get_mut(&connection_token.into())
-			.ok_or(WebSocketError::InvalidConnection(connection_token.0))?;
-		connection.write_message(message)
+		self.send_server_signal(ServerSignal::SendResponse(message, connection_token))
```

Comment:
This was the cause of the deadlock: We sent responses asynchronously to the main event loop in the web-socket server. The fix is to also add responses into the event loop queue and let them be executed from there.

The deadlock could happen, because we have 2 threads that interact: One thread for the web-socket server, that locks connections and handles the RPC calls, which lock the top pool to insert operations. And the second thread which executes operations from the top pool, i.e. also locks the top pool, and then sends responses which also locked connections in the web-socket server.

One downside of this change now, is that a caller of `send_message()` will no longer be informed if the sending itself fails. It will only get the errors until the response is queued in the event queue.

Reply:
I believe that's acceptable, We have not handled these errors anyway, I suppose?

Reply:
No, we just logged the error. So in that regard there is no immediate change in behavior.